src/cdef.rs

```diff
@@ -549,3 +559,750 @@ pub unsafe fn fill(mut tmp: *mut i16, stride: ptrdiff_t, w: c_int, h: c_int) {
+    cost[6] = (cost[6]).wrapping_mul(105 as c_int as c_uint);
+    static div_table: [u16; 7] = [840, 420, 280, 210, 168, 140, 120];
+    let mut n_0 = 0;
+    while n_0 < 7 {
+        let d = div_table[n_0 as usize] as c_int;
+        cost[0] = (cost[0]).wrapping_add(
+            ((partial_sum_diag[0][n_0 as usize] * partial_sum_diag[0][n_0 as usize]
+                + partial_sum_diag[0][(14 - n_0) as usize]
+                    * partial_sum_diag[0][(14 - n_0) as usize])
+                * d) as c_uint,
+        );
+        cost[4] = (cost[4]).wrapping_add(
+            ((partial_sum_diag[1][n_0 as usize] * partial_sum_diag[1][n_0 as usize]
+                + partial_sum_diag[1][(14 - n_0) as usize]
+                    * partial_sum_diag[1][(14 - n_0) as usize])
+                * d) as c_uint,
+        );
+        n_0 += 1;
+    }
+    cost[0] =
+        (cost[0]).wrapping_add((partial_sum_diag[0][7] * partial_sum_diag[0][7] * 105) as c_uint);
+    cost[4] =
+        (cost[4]).wrapping_add((partial_sum_diag[1][7] * partial_sum_diag[1][7] * 105) as c_uint);
+    let mut n_1 = 0;
+    while n_1 < 4 {
+        let cost_ptr: *mut c_uint =
+            &mut *cost.as_mut_ptr().offset((n_1 * 2 + 1) as isize) as *mut c_uint;
+        let mut m = 0;
+        while m < 5 {
+            *cost_ptr = (*cost_ptr).wrapping_add(
+                (partial_sum_alt[n_1 as usize][(3 + m) as usize]
+                    * partial_sum_alt[n_1 as usize][(3 + m) as usize]) as c_uint,
+            );
+            m += 1;
+        }
+        *cost_ptr = (*cost_ptr).wrapping_mul(105 as c_int as c_uint);
+        let mut m_0 = 0;
+        while m_0 < 3 {
+            let d_0 = div_table[(2 * m_0 + 1) as usize] as c_int;
+            *cost_ptr = (*cost_ptr).wrapping_add(
+                ((partial_sum_alt[n_1 as usize][m_0 as usize]
+                    * partial_sum_alt[n_1 as usize][m_0 as usize]
+                    + partial_sum_alt[n_1 as usize][(10 - m_0) as usize]
+                        * partial_sum_alt[n_1 as usize][(10 - m_0) as usize])
+                    * d_0) as c_uint,
+            );
+            m_0 += 1;
+        }
+        n_1 += 1;
+    }
+    let mut best_dir = 0;
+    let mut best_cost: c_uint = cost[0];
+    let mut n_2 = 1;
+    while n_2 < 8 {
+        if cost[n_2 as usize] > best_cost {
+            best_cost = cost[n_2 as usize];
+            best_dir = n_2;
+        }
+        n_2 += 1;
+    }
+    *var = best_cost.wrapping_sub(cost[(best_dir ^ 4 as c_int) as usize]) >> 10;
+    return best_dir;
+}
+
+#[inline(always)]
+#[cfg(all(feature = "asm", any(target_arch = "x86", target_arch = "x86_64"),))]
+unsafe fn cdef_dsp_init_x86<BD: BitDepth>(c: *mut Rav1dCdefDSPContext) {
+    use crate::include::common::bitdepth::BPC;
+
+    let flags = rav1d_get_cpu_flags();
+
+    if !flags.contains(CpuFlags::SSE2) {
+        return;
+    }
+
+    match BD::BPC {
+        BPC::BPC8 => {
+            (*c).fb[0] = dav1d_cdef_filter_8x8_8bpc_sse2;
```

Comment:
```suggestion
            if !flags.contains(CpuFlags::SSSE2) {
                return;
            }

            (*c).fb[0] = dav1d_cdef_filter_8x8_8bpc_sse2;
```

Reply:
By the way, this logic is harder to review/map back to the C when the two bitdepth branches are fully separated, i.e. one `match` vs. a `match` for each `fn` ptr (like how `bd_fn!` does it).  It's fine for now, though.